source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -140,6 +139,12 @@ Http::FilterHeadersStatus FaultFilter::decodeHeaders(Http::HeaderMap& headers, b
   if (headers.EnvoyDownstreamServiceCluster()) {
     downstream_cluster_ =
         std::string(headers.EnvoyDownstreamServiceCluster()->value().getStringView());
+    if (downstream_cluster_.empty()) {
+      downstream_cluster_storage_.reset();
```

Comment:
Filters are not reused between requests so this method is called once. Is this needed?